packages/notebook/src/default-toolbar.tsx

```diff
@@ -70,31 +76,48 @@ export namespace ToolbarItems {
   /**
    * Create save button toolbar item.
    */
-  export function createSaveButton(panel: NotebookPanel): ToolbarButton {
-    return new ToolbarButton({
-      iconClassName: TOOLBAR_SAVE_CLASS + ' foo jp-Icon jp-Icon-16',
-      onClick: () => {
-        if (panel.context.model.readOnly) {
-          return showDialog({
-            title: 'Cannot Save',
-            body: 'Document is read-only',
-            buttons: [Dialog.okButton()]
-          });
-        }
-        panel.context.save().then(() => {
-          if (!panel.isDisposed) {
-            return panel.context.createCheckpoint();
-          }
+  export function createSaveButton(panel: NotebookPanel): Widget {
+    function onClick() {
+      if (panel.context.model.readOnly) {
+        return showDialog({
+          title: 'Cannot Save',
+          body: 'Document is read-only',
+          buttons: [Dialog.okButton()]
         });
-      },
-      tooltip: 'Save the notebook contents and create checkpoint'
-    });
+      }
+      panel.context.save().then(() => {
+        if (!panel.isDisposed) {
+          return panel.context.createCheckpoint();
+        }
+      });
+    }
+    return addToolbarButtonClass(
+      ReactWidget.create(
+        <UseSignal signal={panel.context.fileChanged}>
+          {() => (
+            <ToolbarButtonComponent
+              iconClassName={TOOLBAR_SAVE_CLASS + ' foo jp-Icon jp-Icon-16'}
```

Comment:
Wow, we really add a `foo` class? How did that get in the code?

Reply:
Thanks!